src/main/java/com/okta/tools/OktaAwsCliAssumeRole.java

```diff
@@ -149,6 +158,46 @@ private String getOktaSessionToken() throws IOException {
         }
     }
 
+    private void multiProfileCreateIni() throws  IOException {
+
+        boolean result = false;
+        File okta = new File(System.getProperty("user.home") + "/.okta");
+        File profileIni = new File(System.getProperty("user.home") + "/.okta/.okta-profile-expiry");
+        if (!okta.exists()) {
+            System.out.println("creating Directory and File: " + okta.getName() + profileIni.getName());
+            try {
+                result = okta.mkdir() && profileIni.createNewFile();
+            } catch (SecurityException se) {
+                System.out.println("Permission too low");
+            }
+            if (result)
+                System.out.println("Okta profile expiry created");
+            else
+                System.out.println("Okta profile expiry was not created");
+        }
+    }
+    private boolean multiProfileIniExists() {
+        File okta = new File(System.getProperty("user.home") + "/.okta");
+        return(okta.exists());
+    }
+
+
+    private boolean profileDoesNotExistsOrExpired() throws IOException {
+        Instant start = Instant.now();
+        String profileStore = System.getProperty("user.home") + "/.okta/.okta-profile-expiry";
```

Comment:
This could be shared with the code in multiProfileCreateIni. You could have a method like getMultiProfileIniPath() that works for both methods needed.

Then you can us **profileStore.toFile()** to go from **Path** to **File** and make it work with the **Ini** constructor.